Project-Aurora/Project-Aurora/Devices/YeeLight/YeeLightDevice.cs

```diff
@@ -118,27 +116,43 @@ private int GetFreeTCPPort()
 
         public void Connect(IPAddress lightIP, DoWorkEventArgs token = null)
         {
-            try
+            if (light.IsConnected())
             {
-                if (!light.isConnected())
-                {
-                    IPAddress localIP;
-                    int localListenPort = GetFreeTCPPort(); // This can be any port
-
-                    using (Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, 0))
-                    {
-                        socket.Connect(lightIP, lightListenPort);
-                        localIP = ((IPEndPoint)socket.LocalEndPoint).Address;
-                    }
+                return;
+            }
 
-                    isConnected = light.Connect(lightIP, lightListenPort) && light.SetMusicMode(localIP, localListenPort);
+            //Auto discover a device if the IP is 0.0.0.0
+            if (lightIP.Equals("0.0.0.0"))
+            {
+                List<YeeLightAPI.YeeLightDevice> devices = YeeLightAPI.YeeLightDeviceLocator.DeviceLocator.DiscoverDevices();
+                if (devices.Any())
+                {
+                    light = devices.First();
                 }
+                else
+                {
+                    throw new Exception("IP address is set to 0.0.0.0 and no devices has been located.");
+                }
+            }
+            //If it isn't then set the IP to the provided IP address
+            else
+            {
+                light.SetLightIPAddressAndPort(lightIP, YeeLightAPI.YeeLightConstants.Constants.DefaultCommandPort);
```

Comment:
It's a minor thing, you uses the api constant in below and below the local constant which a little bit confusing for me, because both value is the same. so I rather declare the local cont as private const int lightListenPort = YeeLightAPI.YeeLightConstants.Constants.DefaultCommandPort; and use this local variable after it.

Reply:
I should have named the local variable better, basically there is a local listen port needed for switching the light to music mode which allows us to change color without getting rate limited, and there is a port on the light which we can connect to but be rate limited, we connect to that command port on the light and tell it that our local listen port is [something] and then light opens a connection to our device. This code tries to find a free local port and sends it to the light's command port.